### DISCUSSION TITLE
Error in tests after install

### DISCUSSION THREAD

**Discussion by CicadaDennis** [firedrakeproject/firedrake#2495] [Firedrake support] [answered]
I am attempting an installation of firedrake on an HPC system running the RHEL7 operating system.
The install command I ended up using (after installing petsc separately, because letting firedrake install petsc always failed)

python firedrake-install --disable-ssh --no-package-manager -v --honour-petsc-dir --mpicc=mpicc --mpicxx=mpic++ --mpif90=mpif90 --mpiexec=mpiexec --venv-name=0.13.0 --pip-install="tk mpi4py scipy"

After the installation, as suggested in the installation documentation, I ran the test:
pytest tests/regression/ -k "poisson_strong or stokes_mini or dg_advection"

A lot of errors are printed, but it seems like the main error is:
E       AttributeError: 'petsc4py.PETSc.DMPlex' object has no attribute 'reorderSetDefault'

I would usually associate this sort of error with a version mismatch. 
Is there a particular version of petsc4py that should be used. If so, is there a way to tell the firedrake-install command to install a particular version of petsc4py?

Thanks,
Cicada Dennis
Research Applications and Deep Learning Group
Research Technologies
Indiana University

**Comment by ksagiyam** [comment, marked as answer]
`petsc4py` is now included in the `PETSc` source tree, so they are always compatible. `DMPlexReorderSetDefault()` was introduced in PETSc recently, so I think you need to update your PETSc.

**Reply by CicadaDennis** [reply]
The version of petsc that I am using is a version I just built, it is 3.17.3
That seems to be the most recent version available, from what I can tell, and that is the version under which I had this error occur.
Can you tell me what exact version you have that is working for you?

**Reply by ksagiyam** [reply]
Firedrake is guaranteed to work with the Firedrake fork of PETSc https://github.com/firedrakeproject/petsc, which we try to keep up to date with `petsc/main` (https://gitlab.com/petsc/petsc). `3.17.3` seems already behind `petsc/main`.

**Reply by CicadaDennis** [reply]
3.17.3 is the most recent tagged release available and is shown as having been created 2 weeks ago.
I installed firedrake using the method shown on https://www.firedrakeproject.org/download.html:
curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/master/scripts/firedrake-install
What is the commit # for PETSc I should use that will work with the Firedrake that is installed by that script? Or is there some other method of installation for Firedrake that I should use that will download the Firedrake fork of PETSc that you are referencing?

**Reply by ksagiyam** [reply]
We are often ahead of the latest PETSc release (we follow `petsc/main`, not release).

If you let `firedrake-install` script build PETSc, it will always use the Firedrake fork of PETSc. If you want to build PETSc separately and later do `firedrake-install --honour-petsc-dir ...`, you need to get the Firedrake fork of PETSc with:
`git clone https://github.com/firedrakeproject/petsc` (or by adding `https://github.com/firedrakeproject/petsc` to `remote` and checking out branch `firedrake` in that repo); see `https://github.com/firedrakeproject/petsc/commits/firedrake`.

**Reply by CicadaDennis** [reply]
Thanks for that information. I have had to install PETSc separately, because when I allow firedrake to build it on our system, the build of PETSc fails, so the firedrake install fails. I will download that version of PETSc and hopefully that will work. I will let you know how it goes.

**Reply by CicadaDennis** [reply]
I was able to build it using that petsc (commit 979cc68729db0e8a5f32a051925522944523463f (HEAD -> firedrake, origin/firedrake, origin/HEAD))
And the tests are passing now (mostly).
Thanks again.

**Reply by ksagiyam** [reply]
That sounds good. Thanks.
